### PR TITLE
[JENKINS-33557] - connectivity checks may cause setup wizard to hang for a long time

### PR DESCRIPTION
**Pull request by kzantow**
Issues with internet connectivity can manifest as very long wait times during Jenkins install. This reduces the wait time to 20 seconds and runs all (2) checks in parallel.

@jenkinsci/code-reviewers @reviewbybees esp. @tfennelly 

This addresses: https://issues.jenkins-ci.org/browse/JENKINS-33557

### CODE REVIEW

File: core/src/main/java/hudson/ProxyConfiguration.java [hunk 1]
```diff
@@ -218,22 +224,29 @@ public static ProxyConfiguration load() throws IOException {
...
-                    return new PasswordAuthentication(p.getUserName(),
-                            p.getPassword().toCharArray());
-                }
-            });
+        
+        URLConnection con;
+        if(p==null) {
+            con = url.openConnection();
+        } else {
+            con = url.openConnection(p.createProxy(url.getHost()));
```
Comment (kzantow) [inline core/src/main/java/hudson/ProxyConfiguration.java]: Everything in this else block is just indentation

File: core/src/main/java/hudson/ProxyConfiguration.java [hunk 2]
```diff
@@ -74,6 +74,12 @@
  * @see jenkins.model.Jenkins#proxy
  */
 public final class ProxyConfiguration extends AbstractDescribableImpl<ProxyConfiguration> implements Saveable, Serializable {
+    /**
+     * Holds a default TCP connect timeout set on all connections returned from this class,
+     * note this is value is in milliseconds, it's passed directly to {@link URLConnection#setConnectTimeout(int)}
+     */
+    private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = Integer.getInteger("jenkins.proxy.default_connect_timeout_millis", 20 * 1000);
```
Comment (daniel-beck) [inline core/src/main/java/hudson/ProxyConfiguration.java]: Naming inconsistent with all other System properties we have: https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties

It's camelCase except in the case of a STATIC_FINAL_FIELD_NAME.

File: core/src/main/java/hudson/model/UpdateCenter.java [hunk 3]
```diff
@@ -1132,7 +1134,18 @@ public String getPluginRepositoryBaseUrl() {
 
         private void testConnection(URL url) throws IOException {
             try {
-                Util.copyStreamAndClose(ProxyConfiguration.open(url).getInputStream(),new NullOutputStream());
+                URLConnection connection = (URLConnection) ProxyConfiguration.open(url);
+                connection.setConnectTimeout(20000); // JENKINS-33557 lower connection test timeout (20 sec)
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/UpdateCenter.java]: Is this fixed to 20 seconds deliberately? Why not use reuse the default from `ProxyConfiguration`?

Comment (kzantow) [inline core/src/main/java/hudson/model/UpdateCenter.java]: Ah, hadn't added the `ProxyConfiguration` value at that point, whoops - good catch.

**Pull request merged**